### PR TITLE
DOC : manual improvements

### PR DESCRIPTION
**Pull request by agramfort**
EDIT:
Closes https://github.com/mne-tools/mne-python/issues/3118.

### CODE REVIEW

**Comment by agramfort** [comment]
@jaeilepp or @Eric89GXL please take over

**Comment by jona-sassenhagen** [comment]
Not sure where this should go, but ... I offer to write some manuals. What's important to get done?

**Comment by agramfort** [comment]
@jona-sassenhagen please go over the tutorials we wrote last week.
it surely can be more verbose. For example for you tutorials/plot_eeg_erp.py
is good start.

**Comment by jaeilepp** [comment]
Taking over.

**Comment by jaeilepp** [comment]
Now also includes a fix to ``ica.plot_components``. Matplotlib seems to have a problem with hidden axes.

**Comment by jona-sassenhagen** [comment]
>Now also includes a fix to ica.plot_components. Matplotlib seems to have a problem with hidden axes.

Ah good! This is super annoying.

File: tutorials/plot_eeg_erp.py [hunk 1]
```diff
@@ -183,8 +183,8 @@
 mne.write_evokeds('/tmp/tmp-ave.fif', all_evokeds)
 
 # If Evokeds objects are stored in a dictionary, they can be retrieved by name.
-all_evokeds = dict((cond, epochs[cond]) for cond in event_id)  # dict comp.
-print(all_evokeds["auditory/left"])
+all_evokeds = dict((cond, epochs[cond].average()) for cond in event_id)
+print(all_evokeds['left/auditory'])
```
Comment (jona-sassenhagen) [inline tutorials/plot_eeg_erp.py]: Note, you can always do `all_evokeds['left', 'auditory']` (order doesn't matter).

**Comment by jona-sassenhagen** [comment]
Went over it and I don't see any problems.

appveyor seems unrelated.

**Comment by jaeilepp** [comment]
Maybe it's better to do the improvements in small chunks so the diff doesn't grow too large. This PR includes now fixes to bbox failure in https://github.com/mne-tools/mne-python/issues/3118, broken links and broken examples that caused circle on master to fail.

Ready for review.

**Comment by Eric89GXL** [comment]
@jaeilepp agreed, especially since Circle is currently broken we should get the fixes in ASAP. I'll take a look shortly.

**Comment by Eric89GXL** [comment]
@jaeilepp does this fix #3118?

cc @drammock this fixes some bug in plotting related to hidden axes, might be worth checking to see if there is any way it could also help #3100

**Comment by jaeilepp** [comment]
Yeah. Although it is kind of a workaround. Matplotlib seems to fail on draw, if there are invisible axes. Now we're just hiding the axes by setting the frames invisible.

**Comment by Eric89GXL** [comment]
Workarounds for MPL are fine, we have several of those already IIRC (it's hard to fix the upstream bugs)

**Comment by Eric89GXL** [comment]
Thanks @jaeilepp

**Comment by jona-sassenhagen** [comment]
Thank you @jaeilepp

**Pull request merged**